File: js-benchmarks/batch-benchmarks.js
/**
 * Batch Operations Benchmarks
 */

async function runBatchBenchmarks() {
  await runBenchmarkSuite('BATCH OPERATIONS', [
    {
      name: 'Batch insert 100 elements',
      iterations: 1000,
      fn: benchBatchInsert
    }
  ]);
}

function benchBatchInsert() {
  const container = document.createElement('div');
  const fragment = document.createDocumentFragment();
  
  // Create 100 elements
  for (let i = 0; i < 100; i++) {
    const elem = document.createElement('div');
    elem.className = 'item';
    elem.textContent = `Item ${i}`;
    fragment.appendChild(elem);
  }
  
  // Insert all at once
  container.appendChild(fragment);
}

File: benchmarks/batch_benchmarks.zig
const std = @import("std");
const dom = @import("dom");
const Document = dom.Document;
const Element = dom.Element;

// ============================================================================
// BATCH OPERATIONS BENCHMARKS
// ============================================================================

pub fn benchBatchInsert(allocator: std.mem.Allocator) !void {
    const doc = try Document.init(allocator);
    defer doc.release();

    const parent = try Element.create(allocator, "div");
    _ = try doc.node.appendChild(parent);

    // Insert 100 elements
    var i: usize = 0;
    while (i < 100) : (i += 1) {
        const child = try Element.create(allocator, "div");
        _ = try parent.appendChild(child);
    }
}
